.github/workflows/cli.yml

```diff
@@ -31,55 +42,428 @@ jobs:
         run: |
           cd azure-cli-extensions
           python -m venv .venv
           source .venv/bin/activate
           python -m pip install -U pip
-          pip install azdev
+          pip install azdev pytest
           azdev setup -r .
           azdev extension add apic-extension
 
-      - name: Run tests
+      - name: Discover test cases and set outputs
+        id: discover-tests
         run: |
           cd azure-cli-extensions
           source .venv/bin/activate
-          azdev test apic-extension --live
-        env:
-          USERASSIGNED_IDENTITY: ${{ secrets.USERASSIGNED_IDENTITY }}
-          AWS_ACCESS_KEY_LINK: ${{ secrets.AWS_ACCESS_KEY_LINK }}
-          AWS_SECRET_ACCESS_KEY_LINK: ${{ secrets.AWS_SECRET_ACCESS_KEY_LINK }}
+          cd src/apic-extension
+
+          # Check if specific test cases are provided
+          test_input='${{ inputs.test-cases }}'
+          if [ -n "$test_input" ] && [ "$test_input" != "null" ] && [ "$test_input" != "" ]; then
+            echo "Using specific test cases from input: $test_input"
+            test_cases="$test_input"
+          else
+            echo "Discovering all test cases using pytest..."
+
+            # Use pytest to collect all test cases
+            echo "Debug: Collecting test cases with pytest..."
+
+            # Collect test node IDs and extract just the test function names
+            test_collection=$(python -m pytest --collect-only -q 2>/dev/null | \
+              grep "::test_" | \
+              sed 's/.*::\(test_[^[]*\).*/\1/' | \
+              sort | uniq || true)
+
+            echo "Debug: Raw test collection output:"
+            echo "$test_collection"
+
+            if [ -z "$test_collection" ]; then
+
+              echo "❌ No test cases found with pytest, trying alternative approach..."
+
+              # Fallback: Parse test files directly for test function definitions
+
+              test_collection=$(grep -h "^def test_" $(find . -name "test_*.py") | \
+                sed 's/def \(test_[^(]*\).*/\1/' | \
+                sort | uniq || true)
+
+              echo "Debug: Fallback test collection:"
+              echo "$test_collection"
+            fi
+
+            if [ -z "$test_collection" ]; then
+              echo "❌ No individual test cases found, using full-suite approach"
+              test_cases='["full-suite"]'
+            else
+              test_count=$(echo "$test_collection" | wc -l)
+              echo "✅ Successfully discovered $test_count test cases"
+
+              # Convert to JSON array
+              test_cases=$(echo "$test_collection" | jq -R -s -c 'split("\n") | map(select(length > 0))')
+              total_tests=$(echo "$test_cases" | jq '. | length')
+              echo "Found $total_tests test cases total"
+
+              # Validate JSON format
+              if ! echo "$test_cases" | jq empty 2>/dev/null; then
+                echo "❌ Invalid JSON format, falling back to full-suite"
```

Comment:
Since `test_cases` is serialized by you in line 106, will this situation happen?